File: tests/fixtures/sample-js-project-negative/services/user-service/src/repositories/user.repository.ts
// VIOLATION: architecture/deterministic/circular-module-dependency
// VIOLATION: architecture/deterministic/data-layer-depends-on-api
import { PrismaClient } from '@prisma/client';
import { getUsers } from '../handlers/user.handler';

const prisma = new PrismaClient();

export class UserRepository {
  // VIOLATION: code-quality/deterministic/missing-return-type
  // VIOLATION: code-quality/deterministic/static-method-candidate
  // VIOLATION: code-quality/deterministic/require-await
  async findAll() {
    // VIOLATION: code-quality/deterministic/console-log
    console.log(getUsers);
    return prisma.user.findMany();
  }

  // VIOLATION: code-quality/deterministic/missing-return-type
  // VIOLATION: code-quality/deterministic/static-method-candidate
  // VIOLATION: code-quality/deterministic/require-await
  async findById(id: string) {
    return prisma.user.findUnique({ where: { id } });
  }

  // VIOLATION: code-quality/deterministic/missing-return-type
  // VIOLATION: code-quality/deterministic/static-method-candidate
  // VIOLATION: code-quality/deterministic/require-await
  async create(data: { name: string; email: string }) {
    return prisma.user.create({ data });
  }

  // VIOLATION: code-quality/deterministic/missing-return-type
  // VIOLATION: code-quality/deterministic/static-method-candidate
  // VIOLATION: code-quality/deterministic/require-await
  async delete(id: string) {
    return prisma.user.delete({ where: { id } });
  }
}

File: tests/fixtures/sample-python-project-negative/services/user_service/repositories/user_repository.py
from sqlalchemy import create_engine, select
from sqlalchemy.orm import Session
from ..models.user_model import User
from ..handlers.user_handler import get_users

# VIOLATION: security/deterministic/hardcoded-database-password
engine = create_engine("postgresql://app:secret@localhost:5432/app")


# VIOLATION: style/deterministic/docstring-completeness
class UserRepository:
    # VIOLATION: style/deterministic/docstring-completeness
    # VIOLATION: code-quality/deterministic/missing-type-hints
    def find_all(self) -> list:
        # ARCH-VIOLATION: architecture/deterministic/data-layer-depends-on-api
        # VIOLATION: code-quality/deterministic/console-log
        print(get_users)
        with Session(engine) as session:
            return session.execute(select(User)).scalars().all()

    # VIOLATION: style/deterministic/docstring-completeness
    # VIOLATION: code-quality/deterministic/missing-type-hints
    def find_by_id(self, user_id: str):
        with Session(engine) as session:
            return session.get(User, user_id)

    # VIOLATION: style/deterministic/docstring-completeness
    # VIOLATION: code-quality/deterministic/missing-type-hints
    def create(self, data: dict):
        with Session(engine) as session:
            user = User(name=data["name"], email=data["email"])
            session.add(user)
            session.commit()
            session.refresh(user)
            return user

    # VIOLATION: style/deterministic/docstring-completeness
    def delete(self, user_id: str) -> None:
        with Session(engine) as session:
            user = session.get(User, user_id)
            if user:
                session.delete(user)
                session.commit()
